include/data_fifo.h

```diff
@@ -150,6 +150,18 @@ int data_fifo_num_used_get(struct data_fifo *data_fifo, uint32_t *alloced_num,
  */
 int data_fifo_empty(struct data_fifo *data_fifo);
 
+/**
+ * @brief Deinitialise the data_fifo.
+ *
+ * @note The fifo is emptied first, so it is the users responsibility to release any data items it
```

Comment:
What happens if the user does not release all the data items?

Reply:
What will happen to any buffers that are carried in the message?
We get a message and load it with a pointer to some data (on a slab for example). When all the messages are released what happens to this item and the slab it is on? It may be lost to the system!